IoTDeviceSimulator/web-client/src/DeviceTwin.js

```diff
@@ -146,76 +196,402 @@ export function DeviceTwin(props) {
-        const phase = 0;
-        for (let index = 0; index <= 50; index++) {
-            let tempData = parseFloat(deviceTwin.mean) + Math.sin(index * (2 * Math.PI) / (parseFloat(deviceTwin.sine_period) / 1000) + phase) * parseFloat(deviceTwin.amplitude);
-            let currData = tempData + generateNoise();
+    else if (behaviour === 'Triangular') {
+        for (let i = 0; i < len; i++) {
+            let currData = (2 * parseFloat(amplitude) * Math.asin(Math.sin((2 * Math.PI * (i * (parseFloat(deviceTwin.telemetrySendInterval) / 1000))) / (parseFloat(wave_period) / 1000)))) / Math.PI;
+            ar.push(currData);
+        }
+    }
+    else if (behaviour === 'Sawtooth') {
+        for (let i = 0; i < len; i++) {
+            let currData = 2 * parseFloat(amplitude) * ((i * (parseFloat(deviceTwin.telemetrySendInterval) / 1000)) / (parseFloat(wave_period) / 1000) - Math.floor(1 / 2 + (i * (parseFloat(deviceTwin.telemetrySendInterval) / 1000)) / (parseFloat(wave_period) / 1000)));
             ar.push(currData);
-            arr.push(index);
         }
     }
-    else {
-        for (let index = 0; index <= 50; index++) {
-            let tempData = parseFloat(deviceTwin.mean);
-            let currData = tempData + generateNoise();
+    else if (behaviour === 'Square') {
+        for (let i = 0; i < len; i++) {
+            let currData = parseFloat(amplitude) * Math.sign(Math.sin((2 * Math.PI * (i * (parseFloat(deviceTwin.telemetrySendInterval) / 1000))) / (parseFloat(wave_period) / 1000)));
             ar.push(currData);
-            arr.push(index);
         }
     }
 
-    const isDisabled = (deviceTwin.valueIsBool) ? ((deviceTwin.deviceId !== '' && deviceTwin.deviceName !== '' && deviceTwin.phenomenon !== '' && deviceTwin.telemetrySendInterval !== '') ? false : true) : (deviceTwin.behaviour === 'Sine Function') ? ((deviceTwin.deviceId !== '' && deviceTwin.deviceName !== '' && deviceTwin.phenomenon !== '' && deviceTwin.telemetrySendInterval !== '' && deviceTwin.unit !== '' && deviceTwin.behaviour !== '' && deviceTwin.mean !== '' && deviceTwin.amplitude !== '' && deviceTwin.sine_period !== '' && deviceTwin.noise_magnitude !== '') ? false : true) : ((deviceTwin.deviceId !== '' && deviceTwin.deviceName !== '' && deviceTwin.phenomenon !== '' && deviceTwin.telemetrySendInterval !== '' && deviceTwin.unit !== '' && deviceTwin.behaviour !== '' && deviceTwin.mean !== '' && deviceTwin.noise_magnitude !== '') ? false : true);
+    if (deviceTwin.signalArray && !deviceTwin.valueIsBool) {
+        deviceTwin.currDataArray = [];
+        for (let i = 0; i < len; i++) {
+            deviceTwin.currDataArray[i] = (ar[i] ? ar[i] : 0);
+        }
+        for (let i = 0; i < deviceTwin.signalArray.length; i++) {
+            if (deviceTwin.signalArray[i] !== '') {
+                if (JSON.parse(deviceTwin.signalArray[i])["Behaviour"] === "Sine") {
+                    for (let j = 0; j < len; j++) {

```

Comment:
There seems to be a lot of places where behaviour is compared to check what data to be pushed and which UI to be shown,  can a common function be made to avoid multiple if else comparisions?

Reply:
WIP. 
Committing changes in another PR.